appcues/src/main/java/com/appcues/statemachine/states/BeginningStepState.kt

```diff
@@ -0,0 +1,28 @@
+package com.appcues.statemachine.states
+
+import com.appcues.data.model.Experience
+import com.appcues.statemachine.Action
+import com.appcues.statemachine.Action.RenderStep
+import com.appcues.statemachine.State
+import com.appcues.statemachine.Transition
+
+internal data class BeginningStepState(
+    val experience: Experience,
+    val flatStepIndex: Int,
+) : State {
+
+    override val currentExperience: Experience
+        get() = experience
+    override val currentStepIndex: Int
+        get() = flatStepIndex
+
+    override fun take(action: Action): Transition? {
+        return if (action is RenderStep) {
+            toRenderingStep(action)
+        } else null
+    }
+
+    private fun toRenderingStep(action: RenderStep): Transition {
+        return next(RenderingStepState(experience, flatStepIndex, action.metadata))
```

Comment:
Takes RenderStep(metadata) anda sets it to RenderingStepState, which is the new state the viewModel listens to to know the step it should render.

Reply:
I believe that this is true for a while now. we dont rely on the UI to know we are actually rendering step, so basically the RenderStep action is only possible if we can produceMetada for that step via PresentationEffect, when we go from BeginningToRendering it means that presentingTrait was already called, and we are emiting RenderingStep to the ViewModel, if for some reason the presentingTrait fails we never get to move to RenderingStep. so no analytics is reported.

Reply:
ok, that makes sense - thanks for clarifying.